cpp/src/c_api/neighbor_sampling.cpp

```diff
@@ -63,41 +73,57 @@ struct cugraph_sample_result_t {
 
 namespace {
 
-struct uniform_neighbor_sampling_functor : public cugraph::c_api::abstract_functor {
+struct neighbor_sampling_functor : public cugraph::c_api::abstract_functor {
```

Comment:
Right. I was trying to avoid the redundant codes codes within the 3 functions which is pretty significant. However it makes the code harder to follow as you pointed out. I thought since it is a detail function/functor, it should be okay. I updated the Api to have 3 separate functors.

Reply:
My personal philosophy is that having redundant code that's going to be deleted isn't really redundant.  It can sometimes make things easier, which makes it worth the extra cost (IMO).